system_metrics_collector/src/system_metrics_collector/main.cpp

```diff
@@ -57,54 +69,42 @@ int main(int argc, char ** argv)
     kStatisticsTopicName,
     kDefaultPublishPeriod);
 
+  const auto process_cpu_node =
+    std::make_shared<system_metrics_collector::LinuxProcessCpuMeasurementNode>(
```

Comment:
Any specific reason these are shared and not unique?

Reply:
> Any specific reason these are shared and not unique?

Due to the [executor api](http://docs.ros2.org/eloquent/api/rclcpp/classrclcpp_1_1executor_1_1Executor.html#a393a16cf14b766eeb14334233f10cda7)